Source/Plugins/Core/com.equella.core/js/tsrc/schema/SchemaModule.tsx

```diff
@@ -132,36 +130,15 @@ export const pathForNode = (node: SchemaNode, stripXml = true): string => {
  * @param paths Used by the recursive algorithm to build up the returned list.
  *          When passing in initially, leave as a blank array.
  * @param stripXml Passed into pathForNode. Whether to include the leading /xml.
- *
  */
 export const getAllPaths = (
   nodes: SchemaNode,
   paths: string[],
   stripXml = true
-) => {
-  paths.push(pathForNode(nodes, stripXml));
-  nodes.children?.forEach((childNode) => {
-    paths.concat(getAllPaths(childNode, paths, stripXml));
-  });
-
-  return paths;
-};
-
-/**
- * Recursive helper function that converts a SchemaNode into a corresponding Material UI
- * TreeItem tree.
- * @param nodes The node to generate a TreeItem for.
- */
-export const renderTree = (nodes: SchemaNode) => {
-  return (
-    <TreeItem
-      key={nodes.name}
-      nodeId={pathForNode(nodes, false)}
-      label={nodes.name}
-    >
-      {Array.isArray(nodes.children)
-        ? nodes.children.map((node) => renderTree(node))
-        : null}
-    </TreeItem>
+): string[] => {
+  return paths.concat(pathForNode(nodes, stripXml)).concat(
+    nodes.children?.flatMap((childNode) => {
+      return getAllPaths(childNode, paths, stripXml);
+    }) ?? []
   );
 };
```

Comment:
This can now just have a function body of implicit return types with arrow functions:

```suggestion
): string[] =>
  paths
    .concat(pathForNode(nodes, stripXml))
    .concat(nodes.children?.flatMap((childNode) => getAllPaths(childNode, paths, stripXml))
  ?? [];
```

Reply:
Fixed. Still needs the word return at the start - TS doesn't have that sugar.